### ISSUE TITLE
Error 2488 only shown once

### ISSUE THREAD

**Issue by danny12321** [microsoft/TypeScript#46272]
# Bug Report
When the error ```Type '{type}[] | undefined' must have a '[Symbol.iterator]()' method that returns an iterator.(2488)``` occurs multiple times in a project it only shows once. 

![image](https://user-images.githubusercontent.com/15829978/136526743-a5bbb818-302c-46cb-9419-f64f0291c765.png)
As could be seen in the image above only ```data_1``` has red underline, while I expect to also have red underline by ```data_2```.

### 🕗 Version & Regression Information
When using es5 or older both errors are shown, but from es6 only one error shows.

### ⏯ Playground Link
[Playground link with example](https://www.typescriptlang.org/play?#code/PTAEFcGcEsDsHNQFsCGAXAFgOgE4tgCYD2SoaRoAxjgKbo1kCeADg5GjnPANoC6oRHBEI0AZnBoEAUJSKx2oAuhQAuUO04I+oAD7CCYiQVABeUAFl02PIRIAKAJSgA-KG1rwI8bEkBuKTJyCkpoKAD6AIymblixISi8vqAgoBgokKAooDQ4OILqGEQA7rCB8miKymEATNHcsVjxiclgaRmwFDl5QpCFJVJAA)

### 💻 Code
```ts
// using math.random to create type string[] or undefined
const data: string[] | undefined = Math.random() ? [] : undefined;

const data_1 = [...data]; // has a error shown
const data_2 = [...data]; // has no error shown
```

### 🙁 Actual behavior

The TypeScript compiler is only returning one of the multiple errors when the error is of type 2488.

### 🙂 Expected behavior

The TypeScript compiler should return all the locations of the errors, even if they are the same type. This would be handy for code analysis tools.

**Comment by andrewbranch** [comment]
> The TypeScript compiler should return all the locations of the errors, even if they are the same type.

This is not something we’re interested in doing in general, but this particular example looks surprising to me and feels like _probably_ a mistake.

**Comment by RyanCavanaugh** [comment]
Everything bad we predicted from implementing `ts-ignore` continues to manifest 😕. Biggest regret I have honestly.

**Comment by nicojs** [comment]
Thanks for the response @RyanCavanaugh 🙏. 

Also without the `@ts-expect-error` family of problems here, tools/users might rely on full error reporting. I believe that is also a design goal. 

We at [StrykerJS](https://stryker-mutator.io/docs/stryker-js/introduction/) rely on a full error report in order to validate multiple mutants at the same time in https://github.com/stryker-mutator/stryker-js/pull/3900.
